src/core/server/saved_objects/import/regenerate_ids.ts

```diff
@@ -42,3 +43,39 @@ export const regenerateIds = (objects: SavedObject[]) => {
   }, new Map<string, { id: string; omitOriginId?: boolean }>());
   return importIdMap;
 };
+
+export const regenerateIdsWithReference = async (props: {
+  savedObjects: SavedObject[];
+  savedObjectsClient: SavedObjectsClientContract;
+  workspaces?: string[];
+  objectLimit: number;
+}): Promise<Map<string, { id?: string; omitOriginId?: boolean }>> => {
+  const { savedObjects, savedObjectsClient, workspaces } = props;
+  if (!workspaces || !workspaces.length) {
+    return savedObjects.reduce((acc, object) => {
+      return acc.set(`${object.type}:${object.id}`, { id: object.id, omitOriginId: false });
+    }, new Map<string, { id: string; omitOriginId?: boolean }>());
+  }
+
+  const bulkGetResult = await savedObjectsClient.bulkGet(
```

Comment:
Could we just use the passed `savedObjects` here?

Reply:
We can not, the `savedObjects` here is what in user's import file, the workspaces info may not be the same as what is in kibana index.